tessera-jaxrs/thirdparty-jaxrs/src/main/java/com/quorum/tessera/thirdparty/KeyResource.java

```diff
@@ -0,0 +1,48 @@
+package com.quorum.tessera.thirdparty;
+
+import com.quorum.tessera.admin.ConfigService;
+import com.quorum.tessera.encryption.PublicKey;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+import javax.json.Json;
+import javax.json.JsonArrayBuilder;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Set;
+
+@Api
+@Path("/keys")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class KeyResource {
+
+    private final ConfigService configService;
+
+    public KeyResource(ConfigService configService) {
+        this.configService = configService;
+    }
+
+    @GET
+    @ApiOperation(value = "Fetch local public keys managed by the enclave")
+    @ApiResponses({@ApiResponse(code = 200, message = "Managed public keys")})
+    public Response getPublicKeys() {
+        Set<PublicKey> publicKeys = configService.getPublicKeys();
```

Comment:
I wonder if it is possible to add an extra token with each public key that proves that the specific tessera node answering the request also holds the private part of each key (similar to how it was done in party info).
It would prevent a malicious tessera masquerading as a different node.

Reply:
It would still not be able to receive transactions for the key if the node validation is enabled.

Reply:
But it may cause a third party to connect to the wrong tessera to send an externally signed private transaction (storeraw). I'm not saying this is opening up a big security hole (P2P already validates party info) - just that a malicious node can now do some phishing (especially in a large network).

I also understand that this is meant for cakeshop (and not for 3rd party - external tx signing - users).